### DISCUSSION TITLE
Repository Context, How does emote embedding model provider works?

### DISCUSSION THREAD

**Discussion by rudiservo** [TabbyML/tabby#2595] [Q&A]
I didn't understood how this works honestly, does tabby makes the context, or do we have to make it in the external provider, like llamacpp?

**Comment by wsxiaoys** [comment]
I am about to give a talk on how context providers work internally, focusing on their usage in code completion and answer engines. I will share a version of the presentation deck with the community shortly after it's ready

**Reply by rudiservo** [reply]
I searched and I think I understood what it is, the 3 things I don't understand is:
1- if you use external completion does the embedded still works in tabby?
2 - the scheduler command is gone, how do we manually index?
3 - how do I index with llamacpp?
